### ISSUE TITLE
标签和预测结果可视化

### ISSUE THREAD

**Issue by Deer231** [Linzy0227/SRMNet#5]
你好，我想跟你请教一个问题，你的标签和预测结果的可视化是怎么生成的呢？能不能给一份代码呢

**Comment by Linzy0227** [comment]
预测结果和标签的格式都为.nii.gz格式，通过ITK-SNAP软件查看切片。

**Comment by Deer231** [comment]
好的，谢谢。我看到你predict.py代码中有：
        if savepath:
            pred1 = temp[0, :, :H, :W, :T].cpu().detach().numpy()
            pred1 = pred1.argmax(0)
            assert save_format in ['npy', 'nii']
            if save_format == 'npy':
                np.save(os.path.join(savepath, names[0] + mask_name + '_preds'), pred1)
            if save_format == 'nii':
                if not os.path.exists(os.path.join(savepath, "summit")):
                    os.makedirs(os.path.join(savepath, "summit"))
                oname = os.path.join(savepath, "summit", names[0] + mask_name + '.nii.gz')
                seg_img = np.zeros(shape=(H, W, T), dtype=np.uint8)

                seg_img[np.where(pred1 == 1)] = 1
                seg_img[np.where(pred1 == 2)] = 2
                seg_img[np.where(pred1 == 3)] = 4

                nib.save(nib.Nifti1Image(seg_img, None), oname)
请问您是使用这段代码将预测结果保存为.nii.gz格式的嘛，我在predict.py中使用了您这段代码，可视化出来的切片数是240*240*155，但是数据集和标签是.npy格式的，转换成.nii.gz格式后尺寸大小和切片数和预测结果的尺寸大小和切片数不匹配，请问您是怎么解决这个问题的呢

**Comment by Linzy0227** [comment]
是这个代码，数据集和标签的原始格式也是nii.gz格式的，做可视化分析时用原始数据。

**Comment by Deer231** [comment]
好的好的，真的非常感谢你的回答。还有一个问题就是论文中可视化的图片是通过ITK-SNAP软件查看切片直接截图保存呢，还是使用ITK-SNAP软件转换成图片格式呢

**Comment by Linzy0227** [comment]
软件上有一个相机的标志，可以直接保存png格式的图片。

**Comment by Deer231** [comment]
好的，非常感谢您的解答




YYL
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Linzy0227/SRMNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月3日(星期三) 下午2:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Linzy0227/SRMNet] 标签和预测结果可视化 (Issue #5)





 
软件上有一个相机的标志，可以直接保存png格式的图片。
